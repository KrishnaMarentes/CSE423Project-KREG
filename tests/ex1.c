int main()
{
    int i, j;
    i = 3;
    j = i - 1;
    while(i < 7) {
        if(i < j) {
            ++i;
        } else {
            j = (j + 30) / i;
        }
    }
    return i;
}