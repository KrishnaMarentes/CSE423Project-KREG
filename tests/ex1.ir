function main()
{
int g;
int k;
int KREG.6;
int KREG.7;
int KREG.11;
int KREG.12;
int KREG.13;
int KREG.14;

g = 33;
KREG.6 = 1089;
KREG.7 = 1093;
k = 1093;
goto <KREG.3>;
<KREG.4>:
if g < 6 goto <KREG.8> else goto <KREG.9>;
<KREG.8>:
k = k + 1;
goto <KREG.10>;
<KREG.9>:
KREG.12 = 6 * k;
KREG.13 = KREG.12 + 1;
k = k + KREG.13;
<KREG.10>:
<KREG.3>:
if k < 22 goto <KREG.4>;
<KREG.5>:

KREG.14 = k * 14;
return KREG.14;
}


