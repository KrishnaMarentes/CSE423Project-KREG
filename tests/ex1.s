.section .text

.global main
main:
	push %ebp
	mov %esp, %ebp
	sub $12, %esp
	movl $3, -4(%ebp)
	mov -4(%ebp), %ecx
	mov $1, %edx
	sub %ecx, %edx
	mov %edx, -8(%ebp)
	jmp KREG.3
KREG.4:
	mov -4(%ebp), %ebx
	mov -8(%ebp), %esi
	cmp %ebx, %esi
	jl KREG.7
	jmp KREG.8
KREG.7:
	mov -4(%ebp), %ecx
	mov $1, %edx
	add %ecx, %edx
	mov %edx, -4(%ebp)
	jmp KREG.9
KREG.8:
	mov -8(%ebp), %ecx
	mov $30, %edx
	add %ecx, %edx
	mov %edx, %eax
	mov -4(%ebp), %ecx
	xor %edx, %edx
	div %ecx
	mov %eax, -8(%ebp)
KREG.9:
KREG.3:
	mov -4(%ebp), %ebx
	mov $7, %esi
	cmp %ebx, %esi
	jl KREG.4
KREG.5:
	mov -4(%ebp), %eax
	leave
	ret
