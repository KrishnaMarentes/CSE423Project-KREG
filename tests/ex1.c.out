program 
  declarationList 
    declarationList 
      declarationList 
        declaration 
          structDeclaration structTestStruct{
            unInitVarDecl 
              typeSpecifier int
              unInitVarDeclList 
                varDeclId i
              ;
            unInitVarDecl 
              typeSpecifier int
              unInitVarDeclList 
                varDeclId j
              ;
            unInitVarDecl 
              typeSpecifier int
              unInitVarDeclList 
                varDeclId k
              ;
            };
      declaration 
        funDeclaration 
          typeSpecifier int
          __func(
          params 
            params 
              parameter 
                typeSpecifier int
                paramId a
            ,
            parameter 
              typeSpecifier int
              paramId b
          )
          compoundStmt {
            statementList statementList
              statement 
                returnStmt return
                  expression 
                    simpleExpression 
                      andExpression 
                        unaryRelExpression 
                          relExpression 
                            sumExpression 
                              sumExpression 
                                mulExpression 
                                  unaryExpression 
                                    factor 
                                      mutable a
                              sumop +
                              mulExpression 
                                unaryExpression 
                                  factor 
                                    mutable b
                  ;
            }
    declaration 
      funDeclaration 
        typeSpecifier int
        main(params)
        compoundStmt {
          statementList 
            statementList 
              statementList 
                statementList 
                  statementList 
                    statementList 
                      statementList 
                        statementList 
                          statementList 
                            statementList 
                              statementList 
                                statementList 
                                  statementList 
                                    statementList 
                                      statementList statementList
                                        varDeclaration 
                                          structInit structTestStruct*
                                            varDeclList 
                                              varDeclInitialize 
                                                varDeclId ts
                                                =
                                                expression 
                                                  simpleExpression 
                                                    andExpression 
                                                      unaryRelExpression 
                                                        relExpression 
                                                          sumExpression 
                                                            mulExpression 
                                                              unaryExpression 
                                                                factor 
                                                                  immutable 
                                                                    call malloc(
                                                                      args 
                                                                        argList 
                                                                          expression 
                                                                            simpleExpression 
                                                                              andExpression 
                                                                                unaryRelExpression 
                                                                                  relExpression 
                                                                                    sumExpression 
                                                                                      mulExpression 
                                                                                        unaryExpression 
                                                                                          factor 
                                                                                            immutable 
                                                                                              call sizeof(structTestStruct)
                                                                      )
                                            ;
                                      varDeclaration 
                                        typeSpecifier float
                                        varDeclList 
                                          varDeclInitialize 
                                            varDeclId f
                                            =
                                            expression 
                                              simpleExpression 
                                                andExpression 
                                                  unaryRelExpression 
                                                    relExpression 
                                                      sumExpression 
                                                        mulExpression 
                                                          unaryExpression 
                                                            factor 
                                                              immutable 
                                                                constant 4455.21342
                                        ;
                                    varDeclaration 
                                      typeSpecifier int
                                      varDeclList 
                                        varDeclInitialize 
                                          varDeclId arr_ex[
                                            expression 
                                              simpleExpression 
                                                andExpression 
                                                  unaryRelExpression 
                                                    relExpression 
                                                      sumExpression 
                                                        mulExpression 
                                                          unaryExpression 
                                                            factor 
                                                              immutable 
                                                                constant 3
                                            ]
                                      ;
                                  statement 
                                    expressionStmt 
                                      expression 
                                        mutable 
                                          mutable arr_ex
                                          [
                                          expression 
                                            simpleExpression 
                                              andExpression 
                                                unaryRelExpression 
                                                  relExpression 
                                                    sumExpression 
                                                      mulExpression 
                                                        unaryExpression 
                                                          factor 
                                                            immutable 
                                                              constant 0
                                          ]
                                        =
                                        expression 
                                          simpleExpression 
                                            andExpression 
                                              unaryRelExpression 
                                                relExpression 
                                                  sumExpression 
                                                    mulExpression 
                                                      unaryExpression 
                                                        factor 
                                                          immutable 
                                                            constant 10
                                      ;
                                statement 
                                  expressionStmt 
                                    expression 
                                      mutable 
                                        mutable arr_ex
                                        [
                                        expression 
                                          simpleExpression 
                                            andExpression 
                                              unaryRelExpression 
                                                relExpression 
                                                  sumExpression 
                                                    mulExpression 
                                                      unaryExpression 
                                                        factor 
                                                          immutable 
                                                            constant 1
                                        ]
                                      =
                                      expression 
                                        simpleExpression 
                                          andExpression 
                                            unaryRelExpression 
                                              relExpression 
                                                sumExpression 
                                                  mulExpression 
                                                    unaryExpression 
                                                      factor 
                                                        immutable 
                                                          constant 0x1
                                    ;
                              statement 
                                expressionStmt 
                                  expression 
                                    mutable 
                                      mutable arr_ex
                                      [
                                      expression 
                                        simpleExpression 
                                          andExpression 
                                            unaryRelExpression 
                                              relExpression 
                                                sumExpression 
                                                  mulExpression 
                                                    unaryExpression 
                                                      factor 
                                                        immutable 
                                                          constant 2
                                      ]
                                    =
                                    expression 
                                      simpleExpression 
                                        andExpression 
                                          unaryRelExpression 
                                            relExpression 
                                              sumExpression 
                                                mulExpression 
                                                  unaryExpression 
                                                    factor 
                                                      immutable 
                                                        constant 0x5
                                  ;
                            varDeclaration 
                              typeSpecifier int
                              varDeclList 
                                varDeclInitialize 
                                  varDeclId i
                                  =
                                  expression 
                                    simpleExpression 
                                      andExpression 
                                        unaryRelExpression 
                                          relExpression 
                                            sumExpression 
                                              mulExpression 
                                                unaryExpression 
                                                  factor 
                                                    immutable 
                                                      constant 0
                              ;
                          statement 
                            expressionStmt 
                              expression 
                                mutable i
                                +=
                                expression 
                                  simpleExpression 
                                    andExpression 
                                      unaryRelExpression 
                                        relExpression 
                                          sumExpression 
                                            mulExpression 
                                              unaryExpression 
                                                factor 
                                                  immutable 
                                                    constant 1
                              ;
                        statement 
                          expressionStmt 
                            expression 
                              mutable i
                              <<=
                              expression 
                                simpleExpression 
                                  andExpression 
                                    unaryRelExpression 
                                      relExpression 
                                        sumExpression 
                                          mulExpression 
                                            unaryExpression 
                                              factor 
                                                immutable 
                                                  constant 3
                            ;
                      statement 
                        expressionStmt 
                          expression 
                            mutable i
                            |=
                            expression 
                              simpleExpression 
                                andExpression 
                                  unaryRelExpression 
                                    relExpression 
                                      sumExpression 
                                        mulExpression 
                                          unaryExpression 
                                            factor 
                                              immutable 
                                                constant 2
                          ;
                    statement 
                      expressionStmt 
                        expression 
                          mutable i
                          >>=
                          expression 
                            simpleExpression 
                              andExpression 
                                unaryRelExpression 
                                  relExpression 
                                    sumExpression 
                                      mulExpression 
                                        unaryExpression 
                                          factor 
                                            immutable 
                                              constant 1
                        ;
                  statement 
                    expressionStmt 
                      expression 
                        mutable 
                          mutable ts
                          ->
                          mutable i
                        =
                        expression 
                          simpleExpression 
                            andExpression 
                              unaryRelExpression 
                                relExpression 
                                  sumExpression 
                                    mulExpression 
                                      unaryExpression 
                                        factor 
                                          immutable 
                                            constant 3
                      ;
                statement 
                  expressionStmt 
                    expression 
                      simpleExpression 
                        andExpression 
                          unaryRelExpression 
                            relExpression 
                              sumExpression 
                                mulExpression 
                                  unaryExpression 
                                    factor 
                                      immutable 
                                        call printf(
                                          args 
                                            argList 
                                              argList 
                                                expression 
                                                  simpleExpression 
                                                    andExpression 
                                                      unaryRelExpression 
                                                        relExpression 
                                                          sumExpression 
                                                            mulExpression 
                                                              unaryExpression 
                                                                factor 
                                                                  immutable 
                                                                    constant "Hello, World!            %d\n"
                                              ,
                                              expression 
                                                simpleExpression 
                                                  andExpression 
                                                    unaryRelExpression 
                                                      relExpression 
                                                        sumExpression 
                                                          mulExpression 
                                                            unaryExpression 
                                                              factor 
                                                                mutable 
                                                                  mutable ts
                                                                  ->
                                                                  mutable i
                                          )
                    ;
              statement 
                expressionStmt 
                  expression 
                    simpleExpression 
                      andExpression 
                        unaryRelExpression 
                          relExpression 
                            sumExpression 
                              mulExpression 
                                unaryExpression 
                                  factor 
                                    immutable 
                                      call free(
                                        args 
                                          argList 
                                            expression 
                                              simpleExpression 
                                                andExpression 
                                                  unaryRelExpression 
                                                    relExpression 
                                                      sumExpression 
                                                        mulExpression 
                                                          unaryExpression 
                                                            factor 
                                                              mutable ts
                                        )
                  ;
            statement 
              returnStmt return
                expression 
                  simpleExpression 
                    andExpression 
                      unaryRelExpression 
                        relExpression 
                          sumExpression 
                            mulExpression 
                              unaryExpression 
                                factor 
                                  immutable 
                                    constant 0
                ;
          }
  <EOF>

program 
  structDeclaration structTestStruct
    unInitVarDecl 
      typeSpecifier int
      unInitVarDeclList 
        varDeclId i
    unInitVarDecl 
      typeSpecifier int
      unInitVarDeclList 
        varDeclId j
    unInitVarDecl 
      typeSpecifier int
      unInitVarDeclList 
        varDeclId k
  funDeclaration 
    typeSpecifier int
    __func
    parameter 
      typeSpecifier int
      paramId a
    ,
    parameter 
      typeSpecifier int
      paramId b
    returnStmt return
      sumExpression 
        mutable a
        sumop +
        mutable b
  funDeclaration 
    typeSpecifier int
    main
    varDeclaration 
      structInit structTestStruct*
        varDeclList 
          varDeclInitialize 
            varDeclId ts
            =
            call malloc
              args 
                argList 
                  call sizeofstructTestStruct
    varDeclaration 
      typeSpecifier float
      varDeclList 
        varDeclInitialize 
          varDeclId f
          =
          constant 4455.21342
    varDeclaration 
      typeSpecifier int
      varDeclList 
        varDeclInitialize 
          varDeclId arr_ex[
            constant 3
            ]
    expression 
      mutable 
        mutable arr_ex
        [
        constant 0
        ]
      =
      constant 10
    expression 
      mutable 
        mutable arr_ex
        [
        constant 1
        ]
      =
      constant 0x1
    expression 
      mutable 
        mutable arr_ex
        [
        constant 2
        ]
      =
      constant 0x5
    varDeclaration 
      typeSpecifier int
      varDeclList 
        varDeclInitialize 
          varDeclId i
          =
          constant 0
    expression 
      mutable i
      +=
      constant 1
    expression 
      mutable i
      <<=
      constant 3
    expression 
      mutable i
      |=
      constant 2
    expression 
      mutable i
      >>=
      constant 1
    expression 
      mutable 
        mutable ts
        ->
        mutable i
      =
      constant 3
    call printf
      args 
        argList 
          argList 
            constant "Hello, World!            %d\n"
          ,
          mutable 
            mutable ts
            ->
            mutable i
    call free
      args 
        argList 
          mutable ts
    returnStmt return
      constant 0


